whatrecord/parse.py

```diff
@@ -84,17 +98,26 @@ def parse(
 
     if format in (FileFormat.database, FileFormat.database_definition):
         if format == FileFormat.database_definition or not dbd:
-            return Database.from_file(
-                filename, macro_context=macro_context, version=3 if v3 else 4
+            return Database.from_string(
+                contents,
+                filename=filename,
+                macro_context=macro_context,
+                version=3 if v3 else 4
             )
-        return Database.from_file(
-            filename,
+        return Database.from_string(
+            contents,
+            filename=filename,
             dbd=Database.from_file(dbd, version=3 if v3 else 4),
             macro_context=macro_context
         )
 
     if format == FileFormat.iocsh:
-        md = IocMetadata.from_file(
+        if filename == pathlib.Path("/dev/stdin"):
+            raise ValueError(
+                "IOC shell script parsing not supported through standard input"
+            )
```

Comment:
Can this be merged with the previous file format checks in the earlier block? fail faster and all that jazz

Reply:
Very logical and agreed - addressed in d843fd5